runtime/lua/vim/lsp/util.lua

```diff
@@ -1083,40 +1083,32 @@ function M.locations_to_items(locations)
   for _, d in ipairs(locations) do
     -- locations may be Location or LocationLink
     local uri = d.uri or d.targetUri
-    local fname = assert(vim.uri_to_fname(uri))
     local range = d.range or d.targetSelectionRange
-    table.insert(grouped[fname], {start = range.start})
+    table.insert(grouped[uri], {start = range.start})
   end
 
 
   local keys = vim.tbl_keys(grouped)
   table.sort(keys)
   -- TODO(ashkan) I wish we could do this lazily.
-  for _, fname in ipairs(keys) do
-    local rows = grouped[fname]
-
+  for _, uri in ipairs(keys) do
+    local rows = grouped[uri]
     table.sort(rows, position_sort)
-    local i = 0
-    for line in io.lines(fname) do
-      for _, temp in ipairs(rows) do
-        local pos = temp.start
-        local row = pos.line
-        if i == row then
-          local col
-          if pos.character > #line then
-            col = #line
-          else
-            col = vim.str_byteindex(line, pos.character)
-          end
-          table.insert(items, {
-            filename = fname,
-            lnum = row + 1,
-            col = col + 1;
-            text = line;
-          })
-        end
-      end
-      i = i + 1
+    local bufnr = vim.uri_to_bufnr(uri)
+    if not api.nvim_buf_is_loaded(bufnr) then
+      vim.fn.bufload(bufnr)
+    end
+    for _, temp in ipairs(rows) do
+      local pos = temp.start
+      local row = pos.line
+      local line = (api.nvim_buf_get_lines(bufnr, row, row + 1, true) or {""})[1]
```

Comment:
I think `false` is better here.  I think then you'll always get back the string. I don't remember the exact implementation though, you could check on that though.

Reply:
Changing it to false.


With a buffer:

```

line-1
```

```
lua print(vim.inspect(vim.api.nvim_buf_get_lines(0, 0, 1, false)))  → { "" }
lua print(vim.inspect(vim.api.nvim_buf_get_lines(0, 0, 1, true)))   → { "" }
lua print(vim.inspect(vim.api.nvim_buf_get_lines(0, 1, 2, false)))  → { "line-1" }
lua print(vim.inspect(vim.api.nvim_buf_get_lines(0, 1, 2, true)))   → { "line-1" }
lua print(vim.inspect(vim.api.nvim_buf_get_lines(0, 2, 3, false)))  → { }
lua print(vim.inspect(vim.api.nvim_buf_get_lines(0, 2, 3, true)))   → index out of bounds
```